module/zfs/dsl_deadlist.c

```diff
@@ -957,23 +963,39 @@ dsl_livelist_iterate(void *arg, const blkptr_t *bp, boolean_t bp_freed,
 
 	if ((t != NULL) && (zthr_has_waiters(t) || zthr_iscancelled(t)))
 		return (SET_ERROR(EINTR));
+
+	livelist_entry_t node;
+	node.le_bp = bp;
+	livelist_entry_t *found = avl_find(avl, &node, NULL);
```

Comment:
If we find a matching entry, can we assert that its dedup bit is the same as ours, and if it's dedup'ed its checksum is the same as ours?